File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int low = 0,high = 0,sum = 0;
        int n = nums.length ;
        int res = Integer.MAX_VALUE ;

        while(high<n){
            sum+=nums[high] ;

            while(sum>= target){
                int len = high-low+1;
                res = Math.min(len,res) ;
                sum -= nums[low] ;
                low++;
            }
            high++;
        }
        return (res ==Integer.MAX_VALUE)?0:res ;
        
    }
}

File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.cpp
class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        
        int left=0,right =0,sum =0,mini = INT_MAX,len =0;

        for(right =0;right<nums.size();right++){
            sum+=nums[right] ;

            while(sum>=target){
                len = right-left+1 ;
                mini = min(mini,len) ;
                sum-=nums[left] ;
                left++;

            }

        }
        return (mini== INT_MAX)?0:mini ;


    }
};